Tensile/TensileLibLogicToYaml.py

```diff
@@ -0,0 +1,319 @@
+    
+    #Iterate over Fork parameters and form the Yaml data
+    if args.skipMI != True and isMatrixInsEnabled:
+        forkParameterYamlData = formForkParametersYamlData(currentIndexSolution, MIInstruction9Bits)
+    else:
+        forkParameterYamlData = formForkParametersDataForNonMI(currentIndexSolution)
+        
+    tensileYamlFileData.append(forkParameterYamlData)   
+    
+    #Forms the Library logic string
+    problemSizeYamlData = formProblemSizeYamlData(exactLogic,solutionIndex,scheduleName,deviceNames,architectureName)
+    tensileYamlFileData.append(problemSizeYamlData)
+    
+    #Write the Formed Yaml data into the Yaml File 
+    writeToTensileYamlFile(tensileYamlFile,tensileYamlFileData)
+    
+def formProblemTypeYamlData(currentIndexProblemType, versionString):
+    problemTypeYamlData = []
+    if len(currentIndexProblemType) == 0:
+       raise RuntimeError("Length of problem Type Parameters is empty!!, Please re-check the library logic file !")
+    
+    # Form Global parameters section as well    
+    problemTypeYamlData.append("GlobalParameters:\n")
+    problemTypeYamlData.append("  MinimumRequiredVersion: {}\n\n".format(versionString['MinimumRequiredVersion']))
+    problemTypeYamlData.append("BenchmarkProblems:\n  -\n    - # ProblemType\n")       
+
+    for problemTypeKey, problemTypeValue in currentIndexProblemType.items():
+     # always print DataType, DestDataType, ComputeDataType, HighPrecisionAccumulate, OperationType fields
+      if problemTypeKey == "DataType" or problemTypeKey == "DestDataType" or problemTypeKey == "ComputeDataType":
+         problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, DataType(problemTypeValue).toChar().lower()))
+         continue
+      if (problemTypeKey == "HighPrecisionAccumulate" or problemTypeKey == "OperationType" or 
+          problemTypeKey == "TransposeA" or problemTypeKey == "TransposeB" or problemTypeKey == "UseBeta" or problemTypeKey == "Batched"):
+         problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+         continue
+      if problemTypeKey in defaultProblemType:
+         if problemTypeValue != defaultProblemType[problemTypeKey]:
+            problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+            
+    return ''.join(str(x) for x in problemTypeYamlData)
+            
+def formForkParametersDataForNonMI(currentIndexSolution):
+    forkParametersYamlData = []
+    
+    forkParametersYamlData.append("    - # BenchmarkProblemSizeGroup\n") 
+    forkParametersYamlData.append("      InitialSolutionParameters:\n") 
+    forkParametersYamlData.append("      BenchmarkCommonParameters:\n") 
+    
+    forkParametersYamlData.append("      ForkParameters:\n")
+    macInstruction = ""
+    for forkKey, forkValue in currentIndexSolution.items():
+      if forkKey == "MACInstruction":
+        if forkValue == "MAC": #This is a bug in the library logic file

```

Comment:
You can simplify this part of code.
You just need to change forkValue to "MAD" if it is MAC

if forkValue == "MAC"
  forkValue = "MAD"

Reply:
This may work, but I prefer we fix the library logics to be compatible with the current state of Tensile, rather than tweak this code to compensate for the issue with the MAC. That would be a separate ticket.